tests/benchmark/test_worst_memory.py

```diff
@@ -0,0 +1,336 @@
+"""
+abstract: Tests that benchmark EVMs in the worst-case memory opcodes.
+    Tests that benchmark EVMs in the worst-case memory opcodes.
+
+Tests that benchmark EVMs in the worst-case memory opcodes.
+"""
+
+import pytest
+
+from ethereum_test_base_types.base_types import Bytes
+from ethereum_test_forks import Fork
+from ethereum_test_tools import (
+    Alloc,
+    Bytecode,
+    Environment,
+    StateTestFiller,
+    Transaction,
+)
+from ethereum_test_tools.vm.opcode import Opcodes as Op
+
+from .helpers import code_loop_precompile_call
+
+REFERENCE_SPEC_GIT_PATH = "TODO"
+REFERENCE_SPEC_VERSION = "TODO"
+
+
+class CallDataOrigin:
+    """Enum for calldata origins."""
+
+    TRANSACTION = 1
+    CALL = 2
+
+
+@pytest.mark.valid_from("Cancun")
+@pytest.mark.parametrize(
+    "origin",
+    [
+        pytest.param(CallDataOrigin.TRANSACTION, id="transaction"),
+        pytest.param(CallDataOrigin.CALL, id="call"),
+    ],
+)
+@pytest.mark.parametrize(
+    "size",
+    [
+        pytest.param(0, id="0 bytes"),
+        pytest.param(10, id="10 bytes"),
+        pytest.param(100, id="100 bytes"),
+        pytest.param(1 * 1024, id="1KiB"),
+        pytest.param(10 * 1024, id="10KiB"),
+        pytest.param(100 * 1024, id="100KiB"),
+        pytest.param(1024 * 1024, id="1MiB"),
```

Comment:
@marioevz, I saw your comment in another PR that you might prefer to reduce these range. Is that the case also for this PR?

Maybe since this is a more memory-focused PR instead of 0 and 1MiB we can reduce to 4 cases (e.g., 0, 100, 10KiB and 1MiB) or three (0, 1KiB, 1MiB).

LMK.

Reply:
I think 4 sounds like a good sweet-spot!

Reply:
:+1: